code/game/gamemodes/objective_items.dm

```diff
@@ -27,6 +31,14 @@
 	GLOB.possible_items -= src
 	return ..()
 
+/datum/objective_item/steal/is_valid()
+	if (!length(requiredjob))
+		return TRUE
+	for (var/mob/living/L in GLOB.mob_living_list)
+		if (L.mind?.assigned_role in requiredjob)
+			return TRUE
+	return FALSE
```

Comment:
When are objectives created? If this is done before mobs are created then it will always fail

Reply:
This appears to be fine